scripts/utils/dateUtils.test.ts

```diff
@@ -0,0 +1,141 @@
+import * as cliUtils from './cliUtils'
+import {
+  addDay,
+  convertDaterangeToDateArray,
+  isDateRange,
+  isDateString,
+  testDateArgs,
+  testDateRangeArgs
+} from './dateUtils'
+
+const dateTestCases = [
+  { date: '2020-01-01', expected: true },
+  { date: '2020-01-011', expected: false },
+  { date: '2020-01-af', expected: false },
+  { date: '2020-01-1', expected: false },
+  { date: '2020-1-01', expected: false },
+  { date: '2020-1-1', expected: false },
+  { date: '22-1-1', expected: false },
+  { date: '', expected: false },
+  { date: '2020/01/01', expected: false },
+  { date: '2020/01/1', expected: false },
+  { date: '2020/1/01', expected: false },
+  { date: '2020/1/1', expected: false },
+  { date: '200020/1/1', expected: false },
+  { date: '2022/11111/1', expected: false },
+  { date: '22/1/111111', expected: false }
+]
+
+const daterangeTestCases = [
+  { date: '2020/01/01-2020/01/09', expected: true },
+  { date: '2020/01/01-2020/01/af', expected: false },
+  { date: '2020/01/01-2020/01/099', expected: false },
+  { date: '2020/01/01-2020/019/09', expected: false },
+  { date: '2020/01/01-20209/01/09', expected: false },
+  { date: '2020/01/019-2020/01/09', expected: false },
+  { date: '2020/019/01-2020/01/09', expected: false },
+  { date: '20209/01/01-2020/01/09', expected: false },
+  { date: '2020/01/019-2020/01/099', expected: false },
+  { date: '2020/019/01-2020/019/09', expected: false },
+  { date: '20209/01/01-20209/01/09', expected: false },
+  { date: '2020/01/1-2020/01/09', expected: false },
+  { date: '2020/1/01-2020/01/09', expected: false },
+  { date: '2020/1/1-2020/01/09', expected: false },
+  { date: '2020/01/01-2020/01/9', expected: false },
+  { date: '2020/01/01-2020/1/09', expected: false },
+  { date: '2020/01/01-2020/1/9', expected: false },
+  { date: '2020/01/01-2020/01-09', expected: false },
+  { date: '2020/01/01-2020-01/09', expected: false },
+  { date: '2020/01/01-2020-01-09', expected: false },
+  { date: '2020/01-01-2020/01/09', expected: false },
+  { date: '2020-01/01-2020/01/09', expected: false },
+  { date: '2020-01-01-2020/01/09', expected: false }
+]
```

Comment:
is it worthwhile adding '-' in here for no values provided for date range? happy to be told not needed